go/vt/vttablet/tabletmanager/tm_init.go

```diff
@@ -1003,21 +1010,47 @@ func (tm *TabletManager) initializeReplication(ctx context.Context, tabletType t
 
 	semiSyncAction, err := tm.convertBoolToSemiSyncAction(ctx, reparentutil.IsReplicaSemiSync(durability, currentPrimary.Tablet, tablet))
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
 	if err := tm.fixSemiSync(ctx, tabletType, semiSyncAction); err != nil {
-		return nil, err
+		return "", err
 	}
 
 	// Set primary and start replication.
 	if currentPrimary.Tablet.MysqlHostname == "" {
 		log.Warningf("primary tablet in the shard record does not have mysql hostname specified, possibly because that tablet has been shut down.")
-		return nil, nil
+		return "", nil
+	}
+
+	// Find our own executed GTID set and,
+	// the executed GTID set of the tablet that we are reparenting to.
+	// We will then compare our own position against it to verify that we don't
+	// have an errant GTID. If we find any GTID that we have, but the primary doesn't,
+	// we will not enter the replication graph and instead fail replication.
+	var replicaPos replication.Position
+	replicaPos, err = tm.MysqlDaemon.PrimaryPosition(ctx)
```

Comment:
For consistency and simplicity
```suggestion
	replicaPos, err := tm.MysqlDaemon.PrimaryPosition(ctx)
```

Reply:
Fixed